src/app/cli/src/app.rs

```diff
@@ -243,15 +236,24 @@ pub async fn run(workspace_layout: WorkspaceLayout, args: cli::Cli) -> Result<()
         }
     }
 
+    if let Err(e) = initialize_base_components(&base_catalog).await {
+        tracing::error!(
+            error_dbg = ?e,
+            error = %e.pretty(true),
+            "Initialize base components failed",
+        );
+        return Err(e);
+    }
+
     let is_transactional =
         maybe_db_connection_settings.is_some() && cli_commands::command_needs_transaction(&args);
-    let work_catalog = maybe_server_catalog.unwrap_or(cli_catalog);
+    let work_catalog = maybe_server_catalog.unwrap_or(cli_catalog.clone());
```

Comment:
The transaction should happen on CLI catalog always, as this is what commands inject.
If server is needed, it should be in the chain of CLI catalog.

Reply:
I get the idea.

Done (by rebasing on your changes)